SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/UIWeb/DBPUICommunicationModelScanStateExtensions.swift

```diff
@@ -23,8 +23,9 @@ public extension DBPUIInitialScanState {
 
     init(from brokerProfileQueryData: [BrokerProfileQueryData]) {
         let withoutDeprecated = brokerProfileQueryData.filter { !$0.profileQuery.deprecated }
+        let withoutRemoved = withoutDeprecated.filter { !$0.dataBroker.isRemoved }
 
-        let groupedByBroker = Dictionary(grouping: withoutDeprecated, by: { $0.dataBroker.name }).values
+        let groupedByBroker = Dictionary(grouping: withoutRemoved, by: { $0.dataBroker.name }).values
```

Comment:
I think we should have unit tests that cover this initializer, otherwise it’s difficult to check for potential regression.

Reply:
That’s a good point!